src/ASM/ASMs1D.C

```diff
@@ -1956,15 +1956,14 @@ bool ASMs1D::getNoStructElms (int& n1, int& n2, int& n3) const
 bool ASMs1D::evaluate (const FunctionBase* func, RealArray& values,
                        int, double time) const
 {
-  Go::SplineCurve* scrv = SplineUtils::project(curv,*func,func->dim(),time);
+  std::unique_ptr<Go::SplineCurve> scrv(SplineUtils::project(curv.get(),*func,func->dim(),time));
```

Comment:
Don't see what is gained here by the template usage (except for the more lengthy statement), since this pointer is just a local variable which is deleted before returning.

Reply:
i now immediately see that this pointer won't leak, as opposed to having to look through the code to find the matching delete, and making sure there are not early exits or the likes where it would leak. in this case, sure, there's none of those cases but in general i think that new and delete should be removed from the user facing language and be reserved to low level libraries.